backend/src/tools/http_client.py

```
@@ -0,0 +1,111 @@
+    
+    Args:
+        method: HTTP method (GET, POST, PUT, DELETE)
+        url: Base URL for the API endpoint
+        headers: HTTP headers to include in the request
+        description: Description of what the API call does
+        path_params: Optional dictionary of path parameters to format into the URL
+        
+    Returns:
+        An async function that performs the API call
+    """
+    async def api_call(event: str, data: Dict[str, Any] = {}):
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)